app/hotels/src/allHotels/AllHotelsSearch.js

```diff
@@ -0,0 +1,92 @@
+// @flow
+
+import * as React from 'react';
+import { graphql, createFragmentContainer } from 'react-relay';
+import { PublicApiRenderer } from '@kiwicom/react-native-app-relay';
+import { Layout } from '@kiwicom/react-native-app-common';
+import idx from 'idx';
+
+import SearchForm from './searchForm/SearchForm';
+import FilterStripe from '../filter/FilterStripe';
+import AllHotelsSearchList from './AllHotelsSearchList';
+import type { AllHotelsSearch as AllHotelsSearchData } from './__generated__/AllHotelsSearch.graphql';
+import type { SearchParametersType } from './searchForm/SearchParametersType';
+
+type Props = {|
+  data: AllHotelsSearchData,
+  search: SearchParametersType,
+|};
+
+class AllHotelsSearch extends React.Component<Props> {
+  getCityIdFromData = (data: AllHotelsSearchData): String | null => {
+    const cityId = idx(data, _ => _.edges[0].node.id);
+
+    return cityId ? cityId : null;
+  };
+
+  isReadyToSearch = (): boolean => {
+    const { search: s, data } = this.props;
+
+    return (
+      this.getCityIdFromData(data) && s.checkin !== null && s.checkout !== null
+    );
+  };
+
+  renderInnerComponent = propsFromRenderer => (
+    <AllHotelsSearchList
+      data={propsFromRenderer.allHotels}
+      openSingleHotel={this.props.openSingleHotel}
+    />
+  );
+
+  render() {
+    const { search, onLocationChange, location, data } = this.props;
+
+    return (
+      <Layout>
+        <SearchForm
+          onChange={this.handleSearchChange}
+          onLocationChange={onLocationChange}
+          search={search}
+          location={location}
+          data={null}
+        />
+        <FilterStripe />
+        {this.isReadyToSearch() && (
```

Comment:
I know that you didn't introduce it here - but this is stupid check. The search results may disappear only because you set `checkin > checkout` for a while. This implementation is 💩 ...